localstack/services/dynamodb/provider.py

```diff
@@ -294,6 +294,30 @@ def get_store(context: RequestContext | None = None) -> DynamoDBStore:
     return dynamodb_stores[_account_id][_region]
 
 
+def get_global_table_region(table_name: str, target_region: str | None = None) -> str | None:
+    """
+    Check if the table is a Version 2019.11.21 table.
+    :param target_region: the region we are looking for
+    :param table_name: the name of the table
+    :return: the original region; `None` if this is not a global table
+    """
+    replicas = get_store().REPLICA_UPDATES.get(table_name)
+    if not replicas:
+        return None
+    # todo: consider a different data structure to keep the list of replicas
+    original_region = list(replicas.keys())[0]
+    return original_region if target_region in replicas[original_region] else None
+
+
+def look_for_global_table(
+    context: RequestContext, table_name: str, region: str | None = None
+) -> None:
```

Comment:
in general i would keep method invocations side-effect free if the name suggest that it's just doing a lookup.

maybe a pattern like this would be better:

```python
if global_table_region := find_global_table_region(...):
   context.region = global_table_region
```
and just accept the code duplication, but then it's explicit.

or just rename the function to something that indicates that it may modify the original request.